designdocs/provider_contexts.md

```diff
@@ -0,0 +1,379 @@
+- [Example of Wallet Usage via SIWF](#siwf)
+- [Provider Dashboard Steps](#dashboard)
+
+### **Storage and Structure of the Data** <a id='data'></a>
+
+Currently, the `ProviderRegistryEntry` has only a `name` property, and no way to update said name.
+
+Applications have two different pieces of default and (potentially) internationalized data:
+
+- name
+- logo/image
+
+Limits:
+
+1. Provide a space to internationalize up to 150 localizations (Windows and macOS each have < 150).
+2. Image MUST BE encoded (`png`)[https://www.w3.org/TR/png-3/] that is `250x100` (support for future sizes and light/dark should be considered).
+3. Localization keys should follow BCP 47 format like en-US, fr-FR.
+
+The data structure must support both of these and internationalization.
+
+Proposed are the following changes:
+
+1. Update `ProviderRegistryEntry`.
+
+    Each registry entry supports:
+    - A default name and logo (hash of the logo image).
+    - Localized names and logos (up to 150 locales).
+    - PNG logo support (250x100, up to 128 KiB). Note that the logo is not embedded in the proposal but rather a hash of the logo image.
+    - BCP 47-compliant language codes (e.g., en-US, fr-FR).
+2. Updated `ProviderRegistryEntry` struct have the following properties:
+
+    ```rust
+    type LogoHash = [u8; 32]; // blake2_256 hash of the logo
+    pub struct ProviderRegistryEntry<T: Config> {
+        pub default_name: BoundedVec<u8, T::MaxProviderNameSize>,
+        pub localized_names: BTreeMap<BoundedVec<u8, T::MaxLanguageCodeSize>, BoundedVec<u8, T::MaxProviderNameSize>>,
+        pub default_logo_250_100_png_hash: LogoHash,
+        pub localized_logo_250_100_png_hashes: BTreeMap<BoundedVec<u8, T::MaxLanguageCodeSize>, LogoHash>,
+    }
+    ```
+
+3. New `ProviderToApplicationRegistryEntry` storage be initialized:
+
+    ```rust
+    // Alias for clarity
+    pub type ApplicationIndex = u16;
+
+    #[pallet::storage]
+    pub type ProviderToApplicationRegistryEntry<T: Config> = StorageDoubleMap<
+        _,
+        Twox64Concat,
+        ProviderId,
+        Twox64Concat,
+        ApplicationIndex,
+        ProviderRegistryEntry<T>,
+        OptionQuery,
+    >;
+    ```
+
+4. `MaxProviderNameSize` be increased to `256`.
+5. `MaxProviderLogo250X100Size` be created and the limit set to `131_072` (128 KiB).
+6. Introduce hash-based logo approval mechanism:
+    Logos are not embedded during proposal submission. Instead, a blake2_256 hash of the logo image is included in the proposal.
+    Governance must explicitly approve these logo hashes as part of the provider registration process.
+
+    The approved hashes are recorded in a dedicated storage map:
+
+    ```rust
+        #[pallet::storage]
+        #[pallet::getter(fn approved_logo_hashes)]
+        /// Approved logo hashes for providers.
+        /// This map stores the blake2_256 hash of the logo as the key and an empty tuple as the value.
+        pub ApprovedLogoHashes: StorageMap<
```

Comment:
I don't think we need the `ApprovedLogoHashes` storage do we? Instead the updating of the `ApprovedLogos` requires the path for the lookup in the `ProviderRegistryEntry` right?

Reply:
Agreed, its easier to have it entered in ApprovedLogos and set incoming logos directly against a preset hash